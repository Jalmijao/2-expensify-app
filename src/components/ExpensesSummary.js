import React from 'react';
import { connect } from 'react-redux';
import getExpensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';
import numeral from 'numeral';

export const ExpensesSummary = ({expensesCount,expensesTotal}) => {
    const expensesWord=expensesCount === 1 ? 'expense':'expenses'
    return(
        <div>
            {`Vendo ${expensesCount} ${expensesWord} com um total de ${expensesTotal}`}
        </div>
    );
}


const mapStateToProps = (state) => {
  return {
    expensesTotal: getExpensesTotal(selectExpenses(state.expenses, state.filters)) ,
    expensesCount:state.expenses.length
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
