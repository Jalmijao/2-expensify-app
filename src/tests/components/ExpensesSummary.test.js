import {ExpensesSummary} from '../../components/ExpensesSummary';
import { shallow } from 'enzyme';
import React from 'react';

test('Should render Expenses summary correctly with 1 expense',()=>{
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={333}/>);
    expect(wrapper).toMatchSnapshot();
});

test('Should render Expenses summary correctly with more than 1 expense',()=>{
    const wrapper = shallow(<ExpensesSummary expensesCount={7} expensesTotal={3232322}/>);
    expect(wrapper).toMatchSnapshot();
});