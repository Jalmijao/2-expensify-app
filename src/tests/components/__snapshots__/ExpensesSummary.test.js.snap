// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Expenses summary correctly with 1 expense 1`] = `
<div>
  Vendo 1 expense com um total de 333
</div>
`;

exports[`Should render Expenses summary correctly with more than 1 expense 1`] = `
<div>
  Vendo 7 expenses com um total de 3232322
</div>
`;
